examples/sampleview/core/sampledesigner/IView.h

```diff
@@ -23,14 +23,13 @@ class IView : public QGraphicsObject
 {
     Q_OBJECT
 public:
-    enum { TYPE = DesignerHelper::IVIEW };
     static constexpr qreal basic_width = 200.0;
     static constexpr qreal basic_height = 30.0;
 
-    IView(QGraphicsItem *parent = nullptr);
+    IView(QGraphicsItem *parent, int view_type);
     ~IView() override;
 
-    int type() const override { return TYPE; }
+    int type() const override final { return m_view_type; }
```

Comment:
Is there a way to drop type overload completely from our code, and stick to UserType?